common/src/chain/transaction/signature/inputsig/standard_signature.rs

```diff
@@ -155,7 +125,28 @@ impl StandardInputSignature {
             super::classical_multisig::multisig_partial_signature::SigsVerifyResult::Invalid => return Err(DestinationSigError::InvalidClassicalMultisigAuthorization),
         }
 
-        let serialized_sig = authorization.encode();
+        let input_utxo = inputs_utxos.get(input_num).ok_or(
+            DestinationSigError::InvalidInputUtxoIndex(input_num, inputs_utxos.len()),
+        )?;
+        let is_htlc_spend = input_utxo.map_or(false, |output| match output {
+            TxOutput::Transfer(_, _)
+            | TxOutput::LockThenTransfer(_, _, _)
+            | TxOutput::Burn(_)
+            | TxOutput::CreateStakePool(_, _)
+            | TxOutput::ProduceBlockFromStake(_, _)
+            | TxOutput::CreateDelegationId(_, _)
+            | TxOutput::DelegateStaking(_, _)
+            | TxOutput::IssueFungibleToken(_)
+            | TxOutput::IssueNft(_, _, _)
+            | TxOutput::DataDeposit(_) => false,
+            TxOutput::AtomicSwap(_, _) => true,
+        });
+
+        let serialized_sig = if is_htlc_spend {
+            AuthorizedHashedTimelockContractSpend::Multisig(authorization.encode()).encode()
+        } else {
+            authorization.encode()
+        };
```

Comment:
This has to go.

Please create a separate function for htlc that will call this function.